pyrobosim/pyrobosim/utils/pose.py

```diff
@@ -148,23 +178,26 @@ def from_dict(cls, pose_dict):
                     "roll": eul.get("roll", 0.0),
                 }
             )
+        args["angle_units"] = angle_units
         return cls(**args)
 
     @classmethod
-    def construct(self, data):
+    def construct(self, data, angle_units="radians"):
```

Comment:
Same comment as above, on second thought maybe only the dict case lets you specify units so the code remains simpler.

Reply:
I agree. It also means `construct` method does not need any additional argument as I mentioned above.